packages/client/src/generation/TSClient/Model.ts

```diff
@@ -555,13 +555,13 @@ ${f.name}<T extends ${getFieldArgName(f, name)}<ExtArgs> = {}>(args?: Subset<T,
         name,
       )}<ExtArgs>>): ${getReturnType({
         name: fieldTypeName,
-        actionName: f.outputType.isList ? DMMF.ModelAction.findMany : DMMF.ModelAction.findUnique,
+        actionName: f.outputType.isList ? DMMF.ModelAction.findMany : DMMF.ModelAction.findUniqueOrThrow,
```

Comment:
This is the main part of the fix. The previous code basically said, chaining will depend on `findUnique`, which in turn always adds ` | null` to any result. But we want null or only be present if the root operation is `findUnique` or `findFirst`, or if using `*OrThrow` and chaining on nullable/optional relations.